keystore/keystore.go

```diff
@@ -54,6 +55,14 @@ func NewFSKeystore(dir string) (*FSKeystore, error) {
 	return &FSKeystore{dir}, nil
 }
 
+func (ks *FSKeystore) Has(name string) bool {
+	kp := filepath.Join(ks.dir, name)
+
+	_, err := os.Stat(kp)
```

Comment:
Has should probably also return an error. If we run into some permissions issue, this will report that the key doesnt exist. use `os.IsErrNotExist` to check for file existence, otherwise the error is a real error